src/models/db/fulfillments.js

```diff
@@ -38,15 +42,30 @@ function * insertFulfillments (fulfillments, options) {
 }
 
 function * getFulfillment (transferUuid, options) {
-  return getTransferId(transferUuid, options).then((transferId) => {
-    return db.selectOne({TRANSFER_ID: transferId},
-      options && options.transaction).then((result) => {
-        if (result) {
-          result.transfer_id = transferUuid
+  const transferId = yield getTransferId(transferUuid, options)
+  if (!transferId) {
+    throw new TransferNotFoundError('This transfer does not exist')
+  }
+  const transfer = yield getTransferById(transferId, options)
+  return db.selectOne({TRANSFER_ID: transferId},
+    options && options.transaction).then((result) => {
+      if (!result) {
+        if (transfer.expires_at && moment().isAfter(transfer.expires_at)) {
+          throw new FulfillmentNotFoundError('This transfer expired before it was fulfilled')
         }
-        return result
-      })
-  })
+        throw new FulfillmentNotFoundError('This transfer has no fulfillment')
```

Comment:
Sorry for nit-picking, but could you change it to say "This transfer has not yet been fulfilled"? The current wording sounds very final, even though it's the case where it's not finalized yet